src/components/constructor-form/constructor-form.js

```diff
@@ -0,0 +1,384 @@
+        constructorElement,
+        id:model._id
+      }))
+    }
+  });
+
+  const handleMaterial = (e) => {
+    setFieldValue('material', e.target.value);
+    setMaterialColors(list.filter(el=>el._id===e.target.value)[0].colors);
+  }
+  const handleMaterialColor = (e) => {
+    setFieldValue('color', e.target.value);
+  }
+
+  const checkboxes =(checkBoxName, label)=>[
+    {
+      id: `${checkBoxName}`,
+      dataCy: `${checkBoxName}`,
+      value: values[`${checkBoxName}`],
+      checked: values[`${checkBoxName}`],
+      color: 'primary',
+      label,
+      handler: () => setFieldValue(`${checkBoxName}`, !values[`${checkBoxName}`])
+    }
+  ];
+
+  const handleImageLoad = (e, callback) => {
+    if (e.target.files && e.target.files[0]) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        callback(event);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+    }
+  };
+
+  const handleLoadConstructorImage = (e) => {
+    handleImageLoad(e, (event) => {
+      setFieldValue('image', event.target.result);
+      setConstructorImg(event.target.result);
+    });
+  };
+
+  const textField = (inputValue, inputName, inputLabel, land) => (<TextField
+    data-cy={land?`${land}${inputName}`:`${inputName}`}
+    id={land?`${land}${inputName}`:`${inputName}`}
+    className={styles.textField}
+    variant='outlined'
+    label={inputLabel}
+    value={inputValue}
+    onChange={handleChange}
+    error={touched.inputName && !!errors.inputName}
+  />)
+
+  const selectField = (selectValue,
+    selectChangeAction,
+    selectItemsList,
+    inputLabel,
+    defaultValue) => (<FormControl className={styles.formControl}>
+    <InputLabel id={`multiple-${selectValue}-label`}>
+      {inputLabel}
+    </InputLabel>
+    <Select
+      variant='outlined'
+      labelId={`multiple-${selectValue}-label`}
+      id={`multiple-${selectValue}`}
+      onChange={selectChangeAction}
+      input={<Input />}
+      value={defaultValue || ''}
+      MenuProps={MenuProps}
+      disabled={!selectItemsList.length}
+    >
+      {selectItemsList?selectItemsList.map((selectItem) => (
+        <MenuItem value={selectItem._id} key={selectItem._id}>
+          <div className={styles.selectBox}>
+            {selectValue === 'color' ? <ColorCircle size={SMALL_CIRCLE} color={selectItem.colorHex} /> : null}
+            <span> {selectItem.name[0].value}</span>
+          </div>
+        </MenuItem>
+      )):null}
+    </Select>
+  </FormControl >)
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <CheckboxOptions options={checkboxes('available', config.labels.model.show)} />
+        <CheckboxOptions options={checkboxes('default', config.labels.model.default)} />
+        <Grid item xs={12}>
+          <Paper className={styles.constructorItemUpdate}>
+            <div>
+              <span className={styles.imageUpload}>
+                {config.labels.model.constructorPhoto}
```

Comment:
can it make sense to descructure this?
const { constructorPhoto} = config.labels.model;

Reply:
fixed